### PR TITLE
Fix linter errors

### PR DESCRIPTION
**Pull request by mike-gregory-ovo**
The latest version of `golangci-lint` makes some critical changes to the linters, so I've fixed the various errors across multiple files.

### CODE REVIEW

File: adapters/opsgenie/oncall/oncall_internal_test.go [hunk 1]
```diff
@@ -135,8 +135,8 @@ func TestInit(t *testing.T) {
 				ScheduleID: "schedule",
 			})
 
-			assert.ErrorIs(t, err, gosync.ErrMissingConfig)
-			assert.ErrorContains(t, err, OpsgenieAPIKey)
+			require.ErrorIs(t, err, gosync.ErrMissingConfig)
+			require.ErrorContains(t, err, OpsgenieAPIKey)
```
Comment (jlucktay) [inline adapters/opsgenie/oncall/oncall_internal_test.go:139]: 🚩

File: adapters/opsgenie/schedule/schedule_internal_test.go [hunk 2]
```diff
@@ -360,8 +360,8 @@ func TestInit(t *testing.T) {
 				ScheduleID: "schedule",
 			})
 
-			assert.ErrorIs(t, err, gosync.ErrMissingConfig)
-			assert.ErrorContains(t, err, OpsgenieAPIKey)
+			require.ErrorIs(t, err, gosync.ErrMissingConfig)
+			require.ErrorContains(t, err, OpsgenieAPIKey)
```
Comment (jlucktay) [inline adapters/opsgenie/schedule/schedule_internal_test.go:364]: 🚩

File: adapters/slack/conversation/conversation_internal_test.go [hunk 3]
```diff
@@ -260,8 +259,8 @@ func TestInit(t *testing.T) {
 				Name: "conversation",
 			})
 
-			assert.ErrorIs(t, err, gosync.ErrMissingConfig)
-			assert.ErrorContains(t, err, SlackAPIKey)
+			require.ErrorIs(t, err, gosync.ErrMissingConfig)
+			require.ErrorContains(t, err, SlackAPIKey)
```
Comment (jlucktay) [inline adapters/slack/conversation/conversation_internal_test.go:263]: 🚩

File: adapters/slack/usergroup/usergroup_internal_test.go [hunk 4]
```diff
@@ -242,8 +241,8 @@ func TestInit(t *testing.T) {
 				UserGroupID: "usergroup",
 			})
 
-			assert.ErrorIs(t, err, gosync.ErrMissingConfig)
-			assert.ErrorContains(t, err, SlackAPIKey)
+			require.ErrorIs(t, err, gosync.ErrMissingConfig)
+			require.ErrorContains(t, err, SlackAPIKey)
```
Comment (jlucktay) [inline adapters/slack/usergroup/usergroup_internal_test.go:245]: 🚩

File: adapters/terraformcloud/team/team_internal_test.go [hunk 5]
```diff
@@ -115,17 +116,17 @@ func TestInit(t *testing.T) {
 
 		_, err := Init(ctx, map[gosync.ConfigKey]string{Organisation: "org"})
 
-		assert.ErrorIs(t, err, gosync.ErrMissingConfig)
-		assert.ErrorContains(t, err, Token)
+		require.ErrorIs(t, err, gosync.ErrMissingConfig)
+		require.ErrorContains(t, err, Token)
```
Comment (jlucktay) [inline adapters/terraformcloud/team/team_internal_test.go:120]: ⚠️ Is `Token` going to be an actual token, or the key name to look up a token?

File: adapters/terraformcloud/user/user_internal_test.go [hunk 6]
```diff
@@ -185,8 +185,8 @@ func TestInit(t *testing.T) {
 			Team:         "team",
 		})
 
-		assert.ErrorIs(t, err, gosync.ErrMissingConfig)
-		assert.ErrorContains(t, err, Token)
+		require.ErrorIs(t, err, gosync.ErrMissingConfig)
+		require.ErrorContains(t, err, Token)
```
Comment (jlucktay) [inline adapters/terraformcloud/user/user_internal_test.go:189]: ⚠️

**Review by jlucktay — COMMENTED** [review]
I'm concerned about secrets being leaked through errors and logs and the like.
Apart from that, though, overall good refactor. 👍

**Comment by jlucktay** [comment]
> Hey @jlucktay, those are the names of the expected configuration key e.g. `SlackAPIKey` is https://github.com/ovotech/go-sync/blob/fix-linter-errors/adapters/slack/usergroup/usergroup.go#L56-L59. The ErrorContains is just to ensure that the error contains the name of the key that's failing.

OK, good to know. Those variable names are really alarming in isolation. Even that comment makes me think twice:
> SlackAPIKey is an API key for authenticating with Slack.

To me, that reads like it's describing the secret itself, not the key to look up the secret in another thing somewhere else.

**Comment by mike-gregory-ovo** [comment]
Agreed, bad naming on my behalf. I'm going to leave as-is for now as I imagine this will all change when we move to goplugins anyway!

**Pull request merged**